docs/topics/compatibility-guide-17.md

```diff
@@ -0,0 +1,502 @@
+
+### Prevent implicit coercions between different numeric types
+
+> **Issue**: [KT-48645](https://youtrack.jetbrains.com/issue/KT-48645)
+>
+> **Component**: Kotlin/JVM
+>
+> **Incompatible change type**: behavioral
+>
+> **Short summary**: Kotlin will avoid converting numeric values automatically to a primitive numeric type where only a downcast to that type was needed semantically
+>
+> **Deprecation cycle**:
+>
+> - < 1.5.30: the old behavior in all affected cases
+> - 1.5.30: fix the downcast behavior in generated property delegate accessors,
+>   `-Xuse-old-backend` can be used to temporarily revert to the pre-1.5.30 fix behavior
+> - \>= 1.7.20: fix the downcast behavior in other affected cases
+
+### Deprecate the `enabled` and the `compatibility` modes of the compiler option -Xjvm-default
+
+> **Issue**: [KT-46329](https://youtrack.jetbrains.com/issue/KT-46329)
+>
+> **Component**: Kotlin/JVM
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: Kotlin 1.6.20 warns about the usage of `enabled` and `compatibility` modes of the `-Xjvm-default` compiler option
+>
+> **Deprecation cycle**:
+>
+> - 1.6.20: introduce a warning on the `enabled` and `compatibility` modes of the `-Xjvm-default` compiler option
+> - \>= 1.8.0: raise this warning to an error
+
+### Prohibit calls to functions named `suspend` with a trailing lambda
+
+> **Issue**: [KT-22562](https://youtrack.jetbrains.com/issue/KT-22562)
+>
+> **Component**: Core language
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: Kotlin 1.6 no longer allows calling user functions named `suspend` that have the single argument of a functional type passed as a trailing lambda
+>
+> **Deprecation cycle**:
+>
+> - 1.3.0: introduce a warning on such function calls
+> - 1.6.0: raise this warning to an error
+> - 1.7.0: introduce changes to the language grammar so that `suspend` before `{` is parsed as a keyword
+
+### Prohibit smart cast on base class property if the base class is from another module
+
+> **Issue**: [KT-52629](https://youtrack.jetbrains.com/issue/KT-52629)
+>
+> **Component**: Core language
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: Kotlin 1.7 will no longer allow smart casts on properties declared in another module 
+> that were erroneously allowed before
+>
+> **Deprecation cycle**:
+>
+> - 1.6.0: report a warning on a smart cast of a property declared in the superclass that is in another module
+> - 1.7.0: raise this warning to an error,  
+> `-XXLanguage:-ProhibitSmartcastsOnPropertyFromAlienBaseClass` can be used to temporarily revert to the pre-1.7 behavior
+
+### Do not neglect meaningful constraints during type inference
+
+> **Issue**: [KT-52668](https://youtrack.jetbrains.com/issue/KT-52668)
+>
+> **Component**: Core language
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: Kotlin 1.4–1.6 mistakenly neglected some type constraints during type inference due to an incorrect optimization.
```

Comment:
```suggestion
> **Short summary**: Kotlin 1.4—1.6 neglected some type constraints during type inference due to an incorrect optimization.
```

Reply:
What's wrong with using en-dash for a range? https://www.thepunctuationguide.com/en-dash.html